src/encode.c

```diff
@@ -40,12 +42,16 @@ coap_decode_var_bytes(const uint8_t *buf,unsigned int len) {
 }
 
 unsigned int
-coap_encode_var_bytes(uint8_t *buf, unsigned int val) {
+coap_encode_var_safe(uint8_t *buf, unsigned int length, unsigned int val) {
```

Comment:
As this function usually will be called with `sizeof(...)` for the length argument, we should make it  `size_t`:

```c
coap_encode_var_safe(uint8_t *buf, size_t length, unsigned int val)
```